src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1804,6 +1941,30 @@ def nic_attach_choices(self):
         """
         return self.middleware.call_sync('interface.choices', {'exclude': ['epair', 'tap', 'vnet']})
 
+    @accepts()
+    def pptdev_choices(self):
+        """
+        Available choices for PCI passthru device.
+        """
+        if self.pptdevs is self.NotSet:
```

Comment:
Ok. This was a leftover from the original plan to probe at boot; i.e.,  before actually parsing any vm configuration. I wanted to avoid useless retries when PPT was not used and iommu or pptdevs were not available. Now that the probe is anyway only done when PPT is configured it is probably reasonable to assume that iommu support and pptdev should typically be available and a repeat of the probe may be desirable or at least acceptable.